Source/engine/animationinfo.h

```diff
@@ -83,45 +88,72 @@ class AnimationInfo {
 	 * @param celSprite Pointer to Animation Sprite
 	 * @param numberOfFrames Number of Frames in Animation
 	 * @param ticksPerFrame How many game ticks are needed to advance one Animation Frame
 	 */
-	void ChangeAnimationData(std::optional<CelSprite> celSprite, int numberOfFrames, int ticksPerFrame);
+	void changeAnimationData(std::optional<CelSprite> celSprite, int numberOfFrames, int ticksPerFrame);
 
 	/**
 	 * @brief Process the Animation for a game tick (for example advances the frame)
 	 * @param reverseAnimation Play the animation backwards (for example is used for "unseen" monster fading)
-	 * @param dontProgressAnimation Increase TickCounterOfCurrentFrame but don't change CurrentFrame
+	 * @param dontProgressAnimation Increase TickCounterOfCurrentFrame but don't change currentFrame
 	 */
-	void ProcessAnimation(bool reverseAnimation = false, bool dontProgressAnimation = false);
+	void processAnimation(bool reverseAnimation = false, bool dontProgressAnimation = false);
 
 private:
 	/**
-	 * @brief returns the progress as a fraction (0.0f to 1.0f) in time to the next game tick or 0.0f if the animation is frozen
+	 * @brief Animation sprite
 	 */
-	float GetProgressToNextGameTick() const;
-
+	std::optional<CelSprite> celSprite_;
 	/**
-	 * @brief Specifies how many animations-fractions are displayed between two game ticks. this can be > 0, if animations are skipped or < 0 if the same animation is shown in multiple times (delay specified).
+	 * @brief Current frame of animation
 	 */
-	float TickModifier;
+	int currentFrame_;
 	/**
-	 * @brief Number of game ticks after the current animation sequence started
+	 * @brief Is the animation currently petrified and shouldn't advance with gfProgressToNextGameTick
+	 */
+	bool isPetrified_;
+	/**
+	 * @brief Number of frames in current animation
 	 */
-	float TicksSinceSequenceStarted;
+	int numberOfFrames_;
 	/**
 	 * @brief Animation Frames that will be adjusted for the skipped Frames/game ticks
 	 */
-	int RelevantFramesForDistributing;
+	int relevantFramesForDistributing_;
 	/**
 	 * @brief Animation Frames that wasn't shown from previous Animation
 	 */
-	int SkippedFramesFromPreviousAnimation;
+	int skippedFramesFromPreviousAnimation_;
+	/**
+	 * @brief Increases by one each game tick, counting how close we are to TicksPerFrame
+	 */
+	int tickCounterOfCurrentFrame_;
+	/**
+	 * @brief Specifies how many animations-fractions are displayed between two game ticks. this can be > 0, if animations are skipped or < 0 if the same animation is shown in multiple times (delay specified).
+	 */
+	float tickModifier_;
+	/**
+	 * @brief How many game ticks are needed to advance one Animation Frame
+	 */
+	int ticksPerFrame_;
+	/**
+	 * @brief Number of game ticks after the current animation sequence started
+	 */
+	float ticksSinceSequenceStarted_;
+	/**
+	 * @brief Frame that triggers additional action
+	 */
+	int triggerFrame_;

```

Comment:
I think this is a good idea, but imho we shouldn't add this:
- It makes `AnimationInfo` larger (see #4849)
- It is only relevant for Players and irrelevant, perhaps confusing, for monsters and items (there I disocvered we never changed objects to `AnimationInfo` 😳)
- When changing the player mode, you also have to manually update the trigger frame. If they get out of sync => not good.

Reply:
It is relevant for players AND monsters
I made this update everytime just after set new animation - set in the same function that is called after player/monster mode is changed. So rather no decync should be here.
Go on - click `Files changed` and search for `trigger` to see where it applies.

I understand though, you are trying to minimalize AnimationInfo size.

I also do not like how magmas and storms 2nd trigger frame is actually hardcoded.
I think there could be some better solution to handle this. Perhaps not making this array of trigger frames only just for those 8 type of monsters (+ their bosses)? This 2nd trigger frame of them is not in MonstersData[] ...

Reply:
Sorry my bad. I only did a fast review on the header and didn't notice that github didn't load the diff for `monster.cpp`.

Anyway the other problems stay the same.
Especially the last point concerns me (we have to ensure trigger frame and player/monster mode don't differ).
That's why we also needs the adjustments in `LoadPlayer` or `SyncMonsterAnim`.
Do we know in all places where we check the trigger frame in what player/monster mode we are in? Okay checked again `CanChangeAction` would really be simpler with a triggerFrame.

Another thing to think about is changing how the trigger frames are currently stored.
Before #4221 the check was
```cpp
if (player.AnimInfo.CurrentFrame >= player._pWFrames) {
```
now it's
```cpp
`if (player.AnimInfo.CurrentFrame >= player._pWFrames - 1) {`
```
Cause all frame indexes are zero-based. But we didn't updated the trigger frames. If we reduce all trigger frames by 1, we can have
```cpp
if (player.AnimInfo.CurrentFrame >= player._pWFrames) {
```
back. And I think that's much more readable. What do you think? 🙂 

Perhaps a `bool isAfterTriggerFrame(int8_t triggerFrame)` would help to clariffy it, even if we don't store it in `AnimationInfo`? 🤔 

There are many hardcoded places in diablo. If we could make them more generic/modding friendly that would be great. But I think that's often not a easy task.